plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/deployimage/DeployImageWizardModel.java

```diff
@@ -253,6 +256,14 @@ public void setProject(IProject project) {
 		firePropertyChange(PROPERTY_PROJECT, this.project, this.project = project);
 	}
 
+	public void setProjectOrDefault(IProject project) {
+		if (project != null) {
+			setProject(project);
+		} else if (!projects.isEmpty()) {
+			setProject(projects.get(0));
```

Comment:
In the page, projects are sorted by a comparator, so that projects.get(0) may look as a randomly chosen project. The same problem was solved in NewApplicationWizardModel.getDefaultProject() by applying the comparator.